pageserver/src/tenant.rs

```diff
@@ -2647,7 +2649,11 @@ pub mod harness {
         }
     }
 
+    /// The harness provides a way to create mock certain tenant processes for testing.
+    /// It also creates a temporary dir that serves as a test repository, that gets removed on the harness drop.
     pub struct TenantHarness<'a> {
+        // keep the struct to not to remove tmp dir during the test
+        _temp_repo_dir: TempDir,
```

Comment:
Let's have an explicit `impl Drop for TenantHarness` that does `if !panicking { TempDir::close().expect("") }`. 

I want the test to make noise if it can't clean up

Reply:
Not sure where `panicking` comes from, so just dropping.